### ISSUE TITLE
Migrate from `artifactory_permission_target` to `platform_permission`

### ISSUE THREAD

**Issue by xzhangatlassian** [jfrog/terraform-provider-platform#173]
Hi there,

I  bumpped artifactory terraform provider from 9.8 to the latest 12.5.1. As a result I am forced to migrate to `platform_permission`. I have a permission like

```
resource "platform_permission" "permission-name" {
  name = "permission-name"

  artifact = {
    actions = {
      groups = [
        {
          name        = "migroup"
          permissions = []
        }
      ]
    }

    targets = [
      {
        name = ${module.repo.key}
        include_patterns = ["**"]
      }
    ]
  }
}
```

I ran `terraform import` to import the existing resource into by tf state. However, when I run `terraform plan` terraform output says 

```
          ~ targets = [
              - {
                  - include_patterns = [
                      - "**",
                    ] -> null
                  - name             = "docker-hub-cache" -> null
                },
              + {
                  + include_patterns = [
                      + "**",
                    ]
                  + name             = "docker-hub"
                },
            ]
```

Please notice the repo name will change from `doker-hub-cache` to `docker-hub`. Is that expected? Should I  use the repo key `docker-hub` or the cache repo key `docker-hub-cache` ?

Cheers,
Chandler

**Comment by alexhung** [comment]
@xzhangatlassian Can you confirm you have imported the correct resource? The repo name should not be different if it is from the same permission.

**Comment by xzhangatlassian** [comment]
@alexhung yep, I have imported the right resource. If I change my `platform_permissiom` to 

```
 targets = [
      {
        name = "${module.repo.key}-cache"  #notice the -cache suffix htere
        include_patterns = ["**"]
      }
    ]
```

Terraform plan will not change anything.

**Comment by alexhung** [comment]
@xzhangatlassian Odd. Which version of Artifactory are you using? Are you using SaaS or self-hosted?

I have not seen this behavior in either the old or new API.

**Comment by xzhangatlassian** [comment]
@alexhung I am running artifactory onprem 7.98.7, This only happens to remote to repository (thus the `-cache` suffix). No problem with local repos

**Comment by alexhung** [comment]
@xzhangatlassian Indeed it's remote only. See the doc: https://jfrog.com/help/r/jfrog-artifactory-documentation/remote-repositories. I wonder how the `-cache` repo gets into your permission. It's not something Artifactory exposes to the UI or the API.

Do you see this `docker-hub-cache` repo in your permission list in the UI? Or the permission API response?

**Comment by xzhangatlassian** [comment]
The rest api response contains the `-cache` suffix, for example

```
        "npmjs-cache" : {
          "include_patterns" : [ "**" ],
          "exclude_patterns" : [ ]
        },
```

But not in UI

**Comment by alexhung** [comment]
@xzhangatlassian Whilst you can't create a permission in the UI for the `-cache` repo, you can using the REST API. Is it possible that in the past, something in your system created/updated the permissions with `-cache` repo name suffix?

**Comment by RobinDuhan** [comment]
The older resource used the API `artifactory/api/v2/security/permissions` to create permission targets. This API added the "-cache" suffix to the remote repositories in all of these permission targets, but a GET call to fetch the same permission resulted in output _without_ the cache suffix. Due to this discrepancy, the Terraform state didn't show the "-cache" suffix for its targeted repositories.

The newer resource `platform_permission` uses `/access/api/v2/permissions` - which offers more control over the granularity. And so, fetching the permissions using the new API reveals the "cache" suffix. This, in turn, changes the state file to reflect the target name including the "-cache" suffix.

**You can safely go ahead**. For newer permission targets, this means clear granularity is available. For the older ones being migrated, the name change in the state file is simply reflecting their actual, existing configuration within Artifactory more accurately.

@xzhangatlassian @goodtune Does this makes sense - do you see any other concerns?

**Comment by blabu23** [comment]
I have the same problem here. Artifactory version  7.111.4, platform provider version 2.2.3
Since I try to manage the repositories via GitOps and get the structure via a YAML file from repo admins, the current need to specify '-cache' for the remote repo in the permission is really misleading because the UI does not show the -cache suffix.

So if I do a terraform plan currently, it shows:
```hcl
  # platform_permission.permissions["Write%20remote%20(only%20suseregistry)"] will be updated in-place
  ~ resource "platform_permission" "permissions" {
      ~ artifact = {
          ~ targets = [
              - {
                  - include_patterns = [
                      - "**",
                    ] -> null
                  - name             = "suseregistry-cache" -> null
                },
              + {
                  + include_patterns = [
                      + "**",
                    ]
                  + name             = "suseregistry"
                },
            ]
            # (1 unchanged attribute hidden)
        }
        name     = "Write remote (only suseregistry)"
    }
``` 

Can I apply this without disrupting anything? Or is there still something to be fixed in the provider?
Cheers, Martin

**Comment by RobinDuhan** [comment]
You are good to go.

**Comment by joshuaauger** [comment]
@RobinDuhan, the statement "You are good to go" is misleading. Could we get a definitive answer on the pattern?

When we call the access v2 API, we see both `-cache` and `non-cache` repositories included in permissions. I want to manage these using Terraform, but we have a wrapper module that complicates the process.

I have a few questions:

1. In Terraform, should `platform_permissions` always include only the repository key?
2. Are permissions applied the same way to both `-cache` and `non-cache` repositories? For instance, if I apply the permissions from a `-cache` to the `non-cache` repository key, will it behave the same?
3. If both `-cache` and `non-cache` repositories allow read/write/annotate/delete permissions, what is the difference in scope between the two?
